### DISCUSSION TITLE
*** ERROR *** Cyclic dependency detected in object ordering: burgers <- ini_stress <- burgers

### DISCUSSION THREAD

**Discussion by deyantian18** [idaholab/moose#24999] [Q&A Modules: Solid mechanics]
Dear MOOSE experts,

I created a 2D simple tunnel (small circle + square outside) to model the creep behavior using **Burgers model (kelvin_voigt representation)**, and try to set in-situ stress in the model using **ComputeEigenstrainFromInitialStress** (without this, the model seems works well), 

but if add **ComputeEigenstrainFromInitialStress** it always shows this error:
*** ERROR *** Cyclic dependency detected in object ordering: burgers <- ini_stress <- burgers

 Any suggestion to resolve this issue?

Here is the core block from input file:
```
[Kernels]
  [./TensorMechanics]
    add_variables = true
    strain = SMALL
    planar_formulation = PLANE_STRAIN
    displacements = 'disp_x disp_y'
   # use_displaced_mesh = true
    incremental = true
    volumetric_locking_correction=false
    eigenstrain_names = ini_stress
  [../]
[]

[Materials]
  [./burgers] #kelvin_voigt
    # type = GeneralizedMaxwellModel
    type = GeneralizedKelvinVoigtModel
    creep_modulus = '12000'
    creep_viscosity = '100 18000'
    poisson_ratio = 0.2
    young_modulus = 12000
    driving_eigenstrain=ini_stress
  [../]
  [./stress]
    type = ComputeMultipleInelasticStress
    inelastic_models = 'creep'
    tangent_operator='nonlinear'
   # damage_model= damage
    perform_finite_strain_rotations = false
  [../]
  [./creep]
    type = LinearViscoelasticStressUpdate
  [../]
  [./strain]
    type = ComputeIncrementalSmallStrain
    volumetric_locking_correction=false
    displacements = 'disp_x disp_y'
    eigenstrain_names = ini_stress
  [../]
  [./ini_stress]
    type = ComputeEigenstrainFromInitialStress
    initial_stress = '-5 0 0 0 -5 0 0 0 0'
    eigenstrain_name = ini_stress
  [../]
#  [./ini_stress]
#    type = ComputeEigenstrain
#    eigenstrain_name = ini_stress
#    eigen_base = '-0.00026 0 0  0 -0.00026 0  0 0 0'
 # [../]
[]


[UserObjects]
  [./update]
    type = LinearViscoelasticityManager
    viscoelastic_model = burgers
  [../]
[]
```
Thank you very much for your help.

Best regards,
:D

**Comment by GiudGiud** [comment]
Hello

Converted to discussions until we know for sure it's a bug. 
You could duplicate the stress model to remove the cyclic dependency I think
otherwise @WilkAndy made this object


Guillaume

**Reply by deyantian18** [reply]
Thanks @GiudGiud, could you please elaborate on that? how to duplicate the stress model? thanks, Deyan

**Reply by GiudGiud** [reply]
I m guessing if you set up prefixes for the material properties you can have the same stress model in two blocks. One of them for initialization 

it s a tad hacky but worth a try
